tests/integrationv2/test_buffered_send.py

```diff
@@ -0,0 +1,209 @@
+import pytest
+
+from configuration import available_ports, PROTOCOLS, Protocols, ALL_TEST_CIPHERS, TLS13_CIPHERS, ALL_TEST_CERTS
+from common import ProviderOptions, data_bytes
+from fixtures import managed_process # lgtm [py/unused-import]
+from providers import Provider, S2N, OpenSSL, GnuTLS
+from utils import invalid_test_parameters, get_parameter_name
+
+SEND_DATA_SIZE = 2 ** 16
+
+K_BYTES = 1024 
+SEND_BUFFER_SIZE_MIN = 1031
+SEND_BUFFER_SIZE_MIN_RECOMMENDED = 2 * K_BYTES
+SEND_BUFFER_SIZE_MULTI_RECORD = 17 * K_BYTES
+SEND_BUFFER_SIZE_PREFER_THROUGHPUT = 35 * K_BYTES
+SEND_BUFFER_SIZE_HUGE = 512 * K_BYTES
+
+SEND_BUFFER_SIZES = [
+    SEND_BUFFER_SIZE_MIN_RECOMMENDED,
+    SEND_BUFFER_SIZE_MULTI_RECORD,
+    SEND_BUFFER_SIZE_PREFER_THROUGHPUT,
+]
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("other_provider", [S2N, OpenSSL, GnuTLS], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [S2N], ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", PROTOCOLS, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
```

Comment:
Indeed. I wasn't too sure about the applicability of the certs to the various protocols. I've changed this so that we only use 3 different certs. Investigating it, I've found that, with those three certs all of of the protocols are used:

```
TEST_CERTS = [
    Certificates.RSA_4096_SHA512,
    Certificates.ECDSA_384,
    Certificates.RSA_PSS_2048_SHA256
]
```

This is a great change, because it also allowed me to re-combine the functions that had shared code. (Since there are fewer test cases.)

Is this too minimal of a subset?

Reply:
It's not too minimal. One of each auth type should be sufficient to allow you to use every cipher suite.